pkg/distributor/distributor_test.go

```diff
@@ -4827,16 +4844,15 @@ func TestHandleIngesterPushError(t *testing.T) {
 			if testData.expectedOutputError == nil {
 				require.NoError(t, err)
 			} else {
-				if testData.checkDetails {
-					expectedStat, ok := status.FromError(testData.expectedOutputError)
+				if testData.expectedIngesterPushError {
```

Comment:
Seems weird to have to include a boolean flag in each test case to change the behavior. This test doesn't seem like it involves much setup. Could we instead move the errors which aren't expected to be converted to a push error to another test function to simplify the ways they are verified?

e.g.

```go
t.Run("converable errors", func(t) {
    var pe pushError
    require.ErrorAs(t, err, &pe)
    // other asserts go here
})

t.Run("non-convertable errors", func(t) {
    require.ErrorIs(t, err, tc.expectedOutput)
})
```

Reply:
I have improved this test. I have divided it into 4 categories, corresponding to the 4 cases handled by `Distributor.handleIngesterPushError()`:
- no error
- errors created by `httpgrpc.Errorf()` (that will be removed)
- errors created by `gogo/status` (or `grpc/status`)
- errors without any code (e.g., `context.Canceled`)